crates/miden-tx/src/executor/exec_host.rs

```diff
@@ -166,10 +137,41 @@ where
 
     /// Handles the [`TransactionEvent::EpilogueTxFeeComputed`] and returns an error if the account
     /// cannot pay the fee.
-    fn on_tx_fee_computed(
+    async fn on_tx_fee_computed(
         &self,
         fee_asset: FungibleAsset,
     ) -> Result<Vec<AdviceMutation>, TransactionKernelError> {
+        let asset_witness = self
+            .base_host
+            .store()
+            .get_vault_asset_witness(
+                self.base_host.native_account_header().id(),
+                self.base_host.native_account_header().vault_root(),
+                fee_asset.vault_key(),
+            )
+            .await
+            .map_err(|err| TransactionKernelError::GetVaultAssetWitness {
+                vault_root: self.base_host.native_account_header().vault_root(),
+                vault_key: fee_asset.vault_key(),
+                source: Box::new(err),
+            })?;
+
+        // Find asset in the witness or default to 0 if it isn't present.
+        let initial_native_asset = asset_witness
+            .assets()
+            .find_map(|asset| match asset {
+                Asset::Fungible(fungible_asset)
+                    if fungible_asset.vault_key() == fee_asset.vault_key() =>
+                {
+                    Some(fungible_asset)
+                },
+                _ => None,
+            })
```

Comment:
nit: I would maybe move this into method on `AssetWitness` - e.g., something like:
```rust
impl AssetWitness {
    pub fn get_asset(&self, vault_key: Word) -> Option<Asset> {
        ...
    }
}
```